zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterUtils.java

```diff
@@ -93,22 +97,53 @@ public static TServerSocket createTServerSocket(String portRange)
     throw new IOException("No available port in the portRange: " + portRange);
   }
 
+  // 1) Multiple NetworkCard will be configured on some servers
+  // 2) In the docker container environment, A container will also generate multiple virtual NetworkCard
```

Comment:
Could you add java doc to explain the logic ?
To me, it seems to involves 2 steps:
1. find all available networkcards
2. loo all the neetworkcards, and use the first non-loop ip address.